source/opt/vector_dce.cpp

```diff
@@ -297,6 +299,20 @@ bool VectorDCE::HasScalarResult(const Instruction* inst) const {
   }
 }
 
+uint32_t VectorDCE::GetVectorTypeSize(uint32_t type_id) {
+  analysis::TypeManager* type_mgr = context()->get_type_mgr();
```

Comment:
nit: not sure what the convention is, but `type_mgr` is not needed until after the `if`.

Reply:
Moved after the assert that I added.  We have them at the top fairly often when getting a pointer to the analyses in the ir context.  That way if it is needed earlier, the person will not get it again.